src/main/java/dk/cngroup/lentils/service/ImageService.java

```diff
@@ -0,0 +1,135 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.entity.Hint;
+import dk.cngroup.lentils.entity.Image;
+import dk.cngroup.lentils.entity.formEntity.HintFormObject;
+import dk.cngroup.lentils.exception.DirectoryCanNotBeCreatedException;
+import dk.cngroup.lentils.exception.ImageFileNotFoundException;
+import dk.cngroup.lentils.exception.ImageNotSavedException;
+import dk.cngroup.lentils.exception.ResourceNotFoundException;
+import dk.cngroup.lentils.repository.ImageRepository;
+import org.apache.commons.fileupload.disk.DiskFileItem;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import javax.transaction.Transactional;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+@Service
+public class ImageService {
+
+    private static final String SEPARATOR = File.separator;
+    private static final String IMAGE_DIRECTORY = SEPARATOR + "upload" + SEPARATOR + "images" + SEPARATOR;
```

Comment:
Tohle bych pak nechal jako hodnotu z applicaiton.properties. Ať to jde změnit. Protože na serveru se ty obrázky určitě budou ukládat do  jiné složky.